pkg/trace/api/dogstatsd.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+)
+
+// dogstatsdProxyHandler returns a new HTTP handler which will proxy requests to
+// the DogStatsD endpoint in the Core Agent over UDP or UDS (defaulting to UDS
+// if StatsdSocket is set in the *AgentConfig).
+func (r *HTTPReceiver) dogstatsdProxyHandler() http.Handler {
+	if !r.conf.StatsdEnabled {
+		log.Info("DogstatsD disabled in the Agent configuration")
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, "405 Method Not Allowed", http.StatusMethodNotAllowed)
```

Comment:
Any reason to go with `405` here specifically? Normally I think of `405` as meaning I used GET when I should have used a POST or something? (not super important here)

Reply:
It's what the [appsec endpoint does if appsec isn't enabled](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/appsec/appsec.go#L31), so I was aiming for consistency.

Reply:
Discussed offline. Let's just use 503.